test_vm/tests/withdraw_balance_test.rs

```diff
@@ -0,0 +1,200 @@
+        params: Some(withdraw_balance_params_se),
+        subinvocs: Some(vec![ExpectInvocation {
+            to: beneficiary,
+            method: METHOD_SEND as u64,
+            value: Some(withdraw_amount.clone()),
+            ..Default::default()
+        }]),
+        ..Default::default()
+    }
+    .matches(v.take_invocations().last().unwrap());
+    assert_eq!(withdraw_amount, withdraw_return.amount_withdrawn);
+
+    //withdraw left by beneficiary
+    let left_amount = TokenAmount::from(60);
+    withdraw_amount = TokenAmount::from(100);
+    withdraw_return = withdraw_balance(&v, beneficiary, miner_id, withdraw_amount.clone());
+    withdraw_balance_params_se = serialize(
+        &WithdrawBalanceParams { amount_requested: withdraw_amount },
+        "withdraw  balance params",
+    )
+    .unwrap();
+    ExpectInvocation {
+        to: miner_id,
+        method: MinerMethod::WithdrawBalance as u64,
+        params: Some(withdraw_balance_params_se),
+        subinvocs: Some(vec![ExpectInvocation {
+            to: beneficiary,
+            method: METHOD_SEND as u64,
+            value: Some(left_amount.clone()),
+            ..Default::default()
+        }]),
+        ..Default::default()
+    }
+    .matches(v.take_invocations().last().unwrap());
+    assert_eq!(left_amount, withdraw_return.amount_withdrawn);
+}
+
+#[test]
+fn withdraw_balance_fail() {
+    let store = MemoryBlockstore::new();
+    let mut v = VM::new_with_singletons(&store);
+    let addrs = create_accounts(&v, 3, TokenAmount::from(10_000e18 as i128));
+    let seal_proof = RegisteredSealProof::StackedDRG32GiBV1P1;
+    let (owner, worker, beneficiary, addr) = (addrs[0], addrs[0], addrs[1], addrs[2]);
+
+    // create miner
+    let miner_id = create_miner(
+        &mut v,
+        owner,
+        worker,
+        seal_proof.registered_window_post_proof().unwrap(),
+        TokenAmount::from(1_000e18 as i128),
+    )
+    .0;
+
+    //set as beneficiary
+    let quota = TokenAmount::from(100);
+    let beneficiary_change_proposal = ChangeBeneficiaryParams::new(beneficiary, quota, 500);
+    change_beneficiary(&v, owner, miner_id, &beneficiary_change_proposal);
+
+    let withdraw_amount = TokenAmount::from(100);
+    //message from must be owner
+    apply_code(
+        &v,
+        beneficiary,
+        miner_id,
+        TokenAmount::zero(),
+        MinerMethod::WithdrawBalance as u64,
+        WithdrawBalanceParams { amount_requested: withdraw_amount.clone() },
+        ExitCode::USR_FORBIDDEN,
+    );
+    let withdraw_balance_params_se = serialize(
+        &WithdrawBalanceParams { amount_requested: withdraw_amount },
+        "withdraw  balance params",
+    )
+    .unwrap();
+
+    ExpectInvocation {
+        to: miner_id,
+        method: MinerMethod::WithdrawBalance as u64,
+        params: Some(withdraw_balance_params_se),
+        ..Default::default()
+    }
+    .matches(v.take_invocations().last().unwrap());
```

Comment:
This isn't doing anything useful. It could be useful with an explicit `subinvocs: Some([])` but I think it's clearer to just check the balances afterwards to ensure no transfer happened.

Reply:
looks better after using the withdraw helper that mentioned before
```rs
    withdraw_balance(&v, beneficiary, miner_id, withdraw_amount.clone(), withdraw_amount);
```

Reply:
looks better after using the withdraw helper method that mentioned before 
```rs
withdraw_balance(&v, beneficiary, miner_id, withdraw_amount.clone(), withdraw_amount);
```